src/client/src/notifications.web.ts

```diff
@@ -1,107 +1,88 @@
 import { Subscription } from "rxjs"
 
-import { ExecutionStatus, ExecutionTrade } from "@/services/executions"
+import { ExecutionTrade } from "@/services/executions"
 import { executions$ } from "@/services/executions/executions"
-import { formatNumber } from "@/utils"
 
+import {
+  processCreditAccepted,
+  processCreditQuote,
+  processFxExecution,
+} from "./notificationsUtils"
 import {
   acceptedRfqWithQuote$,
   lastQuoteReceived$,
   QuoteDetails,
   RfqWithQuote,
 } from "./services/credit"
+import { constructUrl } from "./utils/url"
 
-const icon =
-  navigator.userAgent.indexOf("Chrome") !== -1 &&
-  navigator.userAgent.indexOf("Win") !== -1
-    ? "./static/media/reactive-trader-icon-no-bkgd-256x256.png"
-    : "./static/media/reactive-trader-icon-dark-256x256.png" // MacOS & Firefox notifications have white backgrounds, so use dark backgrounded icon
-
-const sendFxTradeNotification = (executionTrade: ExecutionTrade) => {
-  const notification = {
-    ...executionTrade,
-    valueDate: executionTrade.valueDate.toString(),
-    tradeDate: executionTrade.valueDate.toString(),
-  }
+const fxIconUrl = constructUrl("/static/media/reactive-trader-fx.svg")
+const creditIconUrl = constructUrl("/static/media/reactive-trader-credit.svg")
 
-  const status =
-    notification.status === ExecutionStatus.Done ? "Accepted" : "Rejected"
-  const title = `Trade ${status}: ID ${notification.tradeId}`
-  const body = `${notification.direction} ${
-    notification.dealtCurrency
-  } ${formatNumber(
-    notification.notional,
-  )} vs ${notification.currencyPair.substr(3)} @ ${notification.spotRate}`
+const sendFxTradeNotification = (trade: ExecutionTrade) => {
+  const { title, tradeDetails: tradeCurrencyDetails } =
+    processFxExecution(trade)
 
   const options: NotificationOptions = {
-    body,
-    icon,
+    body: `${trade.direction} ${tradeCurrencyDetails}`,
+    icon: fxIconUrl,
     dir: "ltr",
-    data: notification,
-    // tag: "trade", TODO: investigate why this field causes malfunctions on certain versions of chrome
   }
 
-  new Notification(title, options)
+  const fxNote = new Notification(title, options)
+  fxNote.onclick = (notification) => console.log(notification)
```

Comment:
Did you leave this intentionally?

Reply:
nope 🤦🏻  ... a test for onClick, works on desktop but not iOS unfortunately -- will remove

Reply:
might use this in future for OpenFin-like behaviour - obv atm this is just simple blotter highlighting